### ISSUE TITLE
[BUG] ClamAV is broken

### ISSUE THREAD

**Issue by DerOetzi** [docker-mailserver/docker-mailserver#2685]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

clamav

### What happened and when does this occur?

```Markdown
On each mail received there are following logs since upgrade to 11.1.0 and Mails seems not to be checked by clamav:


mailserver      | Jul 19 21:30:41 25b00dd3ad13 amavis[4479]: (04479-01) (!)connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
mailserver      | Jul 19 21:30:42 25b00dd3ad13 amavis[4479]: (04479-01) (!)connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
mailserver      | Jul 19 21:30:42 25b00dd3ad13 amavis[4479]: (04479-01) (!)ClamAV-clamd: All attempts (1) failed connecting to /var/run/clamav/clamd.ctl, retrying (2)
mailserver      | Jul 19 21:30:48 25b00dd3ad13 amavis[4479]: (04479-01) (!)connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
mailserver      | Jul 19 21:30:48 25b00dd3ad13 amavis[4479]: (04479-01) (!)ClamAV-clamd av-scanner FAILED: run_av error: Too many retries to talk to /var/run/clamav/clamd.ctl (All attempts (1) failed connecting to /var/run/clamav/clamd.ct
l) at (eval 95) line 659.\n
mailserver      | Jul 19 21:30:48 25b00dd3ad13 amavis[4479]: (04479-01) (!)WARN: all primary virus scanners failed, considering backups
```
```


### What did you expect to happen?

```Markdown
Incoming mails should be checked by clamav
```

### How do we replicate the issue?

```Markdown
1. The only change I did is to upgrade docker image to 11.1.0
```

### DMS version

v11.1.0

### What operating system is DMS running on?

Linux

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker Compose

### Relevant log output

```bash
mailserver      | Jul 19 21:30:41 25b00dd3ad13 amavis[4479]: (04479-01) (!)connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
mailserver      | Jul 19 21:30:42 25b00dd3ad13 amavis[4479]: (04479-01) (!)connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
mailserver      | Jul 19 21:30:42 25b00dd3ad13 amavis[4479]: (04479-01) (!)ClamAV-clamd: All attempts (1) failed connecting to /var/run/clamav/clamd.ctl, retrying (2)
mailserver      | Jul 19 21:30:48 25b00dd3ad13 amavis[4479]: (04479-01) (!)connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
mailserver      | Jul 19 21:30:48 25b00dd3ad13 amavis[4479]: (04479-01) (!)ClamAV-clamd av-scanner FAILED: run_av error: Too many retries to talk to /var/run/clamav/clamd.ctl (All attempts (1) failed connecting to /var/run/clamav/clamd.ct
l) at (eval 95) line 659.\n
mailserver      | Jul 19 21:30:48 25b00dd3ad13 amavis[4479]: (04479-01) (!)WARN: all primary virus scanners failed, considering backups
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [ ] I am inexperienced with mail servers
- [ ] I am uncomfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

**Comment by DerOetzi** [comment]
On deeper research I found this log message as well:

```
mailserver      | Jul 19 21:42:20 654355fe8e12 postfix/local[4837]: 1DFE51207A6: to=<clamav@xxxxxxxxxxxxxx.xx>, orig_to=<clamav>, relay=local, delay=294129, delays=294129/0.02/0/0.01, dsn=4.2.0, status=deferred (cannot update mailbox /var
/mail/clamav for user clamav. destination /var/mail/clamav is not owned by recipient)
```

**Comment by kladiv** [comment]
Hello @DerOetzi,
i hit the same issue in K8S (GKE)

```
connect to /var/run/clamav/clamd.ctl failed, attempt #1: Can't connect to a UNIX socket /var/run/clamav/clamd.ctl: No such file or directory
```

After some troubleshooting and analysis i found that the **clamd** process fails to start properly due to OOM (out of memory) and the current `ExitOnOOM false` parameter. ClamAV requires more memory to complete the startup.

I suggest the following solution:
- increase the limit resources of the container. A good value is 2Gi or higher
- set the parameters below in */etc/clamav/clamd.conf*:
  -  `ConcurrentDatabaseReload false` (according to this [link](https://betatim.github.io/posts/clamav-memory-usage/))
  - `ExitOnOOM true`. So if clamd process get killed by [OOM Killer](https://www.kernel.org/doc/gorman/html/understand/understand016.html) (inside the container), it can be automatically restarted by supervisord (that is used by docker-mailserver to manage application processes like clamav, dovecot, amavis, postfix, etc.).  
    To apply these parameters add lines below into script **user-patches.sh** script:
    ```bash
    sed -i -e '/^ExitOnOOM/d' -e '$a ExitOnOOM true' /etc/clamav/clamd.conf
    sed -i -e '/^ConcurrentDatabaseReload/d' -e '$a ConcurrentDatabaseReload false' /etc/clamav/clamd.conf
    ```

I hope this can help someone (or maybe can be added to FAQs)

**Comment by chongma** [comment]
I am getting clamd sticking on high cpu after recent image upgrade

**Comment by georglauterbach** [comment]
> I am getting clamd sticking on high cpu after recent image upgrade

Have you properly shut down the container? Please user `docker compose down` and `docker compose up`  if you haven't already. Otherwise, please file a new report.
